File: kotlin/src/main/kotlin/io/github/jkjamies/automationagent/agent/covfixer/Coverage.kt
/*
 * Package covfixer is the test-coverage configuration of the fixflow engine: it triages an agnostic
 * coverage report into source files with meaningful uncovered logic, then generates language-aware
 * tests for them. Its prompts are entirely separate from the lint-fixer's; only the deterministic
 * loop is shared (fixflow).
 */
package io.github.jkjamies.automationagent.agent.covfixer

import io.github.jkjamies.automationagent.agent.fixflow.Deps
import io.github.jkjamies.automationagent.agent.fixflow.Engine
import io.github.jkjamies.automationagent.agent.fixflow.Spec
import io.github.jkjamies.automationagent.agent.setup.Prompts

internal val prompts = Prompts.forAgent("covfixer")

/** Builds the coverage-fixer engine. */
fun newEngine(deps: Deps): Engine =
    Engine(
        Spec(
            name = "coverage",
            branch = "automation-agent/test-coverage",
            label = "automation-agent-coverage",
            checkName = "agent-coverage-verify",
            commitMessage = "automation-agent: add test coverage",
            prTitle = "automation-agent: add test coverage",
            successTitle = "Coverage fix succeeded ✅",
            reviewTitle = "Coverage fix needs human review ⚠️",
            triage = ::triage,
            analyze = ::analyze,
        ),
        deps,
    )

File: go/internal/agent/covfixer/coverage.go
// Package covfixer is the test-coverage configuration of the fixflow engine: it
// triages an agnostic coverage report into source files with meaningful uncovered
// logic, then generates language-aware tests for them. Its prompts are entirely
// separate from the lint-fixer's; only the deterministic loop is shared (fixflow).
package covfixer

import (
	"embed"

	"automation-agent/internal/agent/fixflow"
	"automation-agent/internal/agent/setup"
)

//go:embed prompts/*.md
var promptFS embed.FS

var prompts = setup.NewPrompts(promptFS)

// NewEngine builds the coverage-fixer engine.
func NewEngine(d fixflow.Deps) *fixflow.Engine {
	return fixflow.NewEngine(fixflow.Spec{
		Name:          "coverage",
		Branch:        "automation-agent/test-coverage",
		CheckName:     "agent-coverage-verify",
		CommitMessage: "automation-agent: add test coverage",
		PRTitle:       "automation-agent: add test coverage",
		SuccessTitle:  "Coverage fix succeeded ✅",
		ReviewTitle:   "Coverage fix needs human review ⚠️",
		Triage:        Triage,
		Analyze:       Analyze,
	}, d)
}
